### PR TITLE
MAINT: ndimage: malloc fail check 

### PR DESCRIPTION
**Pull request by clementkng**
Added malloc checks for returning null based on issue #9999 for the `ni_morphology.c`.

### CODE REVIEW

File: scipy/ndimage/src/ni_morphology.c [hunk 1]
```diff
@@ -623,6 +623,10 @@ int NI_DistanceTransformBruteForce(PyArrayObject* input, int metric,
             temp->index = jj;
             temp->coordinates = malloc(PyArray_NDIM(input) *
                                        sizeof(npy_intp));
+            if (!temp->coordinates) {
+                PyErr_NoMemory();
```
Comment (pv) [inline scipy/ndimage/src/ni_morphology.c]: `temp` needs to be freed here.

**Review by pv — CHANGES_REQUESTED** [review]

File: scipy/ndimage/src/ni_morphology.c (hunk 1 above)
Comment (peterbell10) [reply scipy/ndimage/src/ni_morphology.c]: `temp` is assigned to `border_elements` above which is singly linked list. So, it was fine as it was before.

Comment (pv) [reply scipy/ndimage/src/ni_morphology.c]: Ok, indeed, it shouldn't be freed here.

Comment (clementkng) [reply scipy/ndimage/src/ni_morphology.c]: I reverted the old commit.

**Review by peterbell10 — APPROVED** [review]

**Comment by larsoner** [comment]
Sounds like @pv also agreed with this changeset so I'll merge, thanks @clementkng

**Pull request merged**